### ISSUE TITLE
Limit number of updates per groupId

### ISSUE THREAD

**Issue by anilkumarmyla** [fthomas/scala-steward#1068]
This is similar to #1046 but applies the limiting per groupId rather than at a repo level.

This will be useful in projects which amalgamate a bunch of modules from the same groupId, which usually happens in a corp setting - team creates a bunch of low level modules under the same groupId and combines all of them in a big project. If there is rapid development in the low level modules this creates a lot of pr noise in the big project, limiting the number of updates per groupId would help in such a case. 

Let me know if this makes sense or any questions you have

**Comment by anilkumarmyla** [comment]
Also the explanation here is the primary driver for #412

**Comment by manuelcueto** [comment]
#1046 works on limiting the number of updates, so if you have 1 update that is applied to a group id it would work here. eg: 
```scala
Seq("libary1","libary2","libary3","libary4").map("com.foo" % _ % "1.2")
```
you would only get 1 pr updating all of them. obviously this would only work if all your modules share the version. 
or are you actually thinking of a scenario like : 
```scala
"org.icecream" % "libary1" % "1.1",
"org.icecream" % "libary2" % "2.2",
"org.icecream" % "libary3" % "3.3"
```
And then providing a limit of 2 upgrades on "org.icecream" 
so you wouldn't update library 3?

**Comment by anilkumarmyla** [comment]
>And then providing a limit of 2 upgrades on "org.icecream"
>so you wouldn't update library 3?

Yes, this is the behavior we would want. This also heavily relies on scala-steward running frequently, so the next time it runs it would update library 3 given upgrades to library 1 & 2 are merged
